rmw_connext_cpp/src/functions.cpp

```diff
@@ -198,6 +199,17 @@ rmw_create_publisher(
     return NULL;
   }
 
+  switch (qos_policy) {
+    case RMW_QOS_POLICY_BEST_EFFORT:
+      datawriter_qos.reliability.kind = DDS_BEST_EFFORT_RELIABILITY_QOS;
+      break;
+    case RMW_QOS_POLICY_RELIABLE:
+      datawriter_qos.reliability.kind = DDS_RELIABLE_RELIABILITY_QOS;
+      break;
+    default:
+      throw std::runtime_error("Unknown QoS policy");
```

Comment:
It would be good if the error string would include the actual numeric value of `qos_policy`.

Same below.